core/templates/cowdata.h

```diff
@@ -260,26 +269,27 @@ typename CowData<T>::USize CowData<T>::_copy_on_write() {
 		/* in use by more than me */
 		USize current_size = *_get_size();
 
-		USize *mem_new = (USize *)Memory::alloc_static(_get_alloc_size(current_size) + ALLOC_PAD, false);
-		mem_new += 2;
+		uint8_t *mem_new = (uint8_t *)Memory::alloc_static(_get_alloc_size(current_size) + _data_offset, false);
+		ERR_FAIL_NULL_V(mem_new, 0);
 
-		new (mem_new - 2) SafeNumeric<USize>(1); //refcount
-		*(mem_new - 1) = current_size; //size
+		SafeNumeric<USize> *_refc_ptr = (SafeNumeric<USize> *)(mem_new + _refc_offset);
+		USize *_size_ptr = (USize *)(mem_new + _size_offset);
+		T *_data_ptr = (T *)(mem_new + _data_offset);
```

Comment:
I'd suggest helper methods for these three calculations, just to avoid confusion and errors, as they're used a few times over

Reply:
Added helpers.